doc/developer/feature-branch.rst

```diff
@@ -0,0 +1,109 @@
+- Feature branch is not linked to CI/CD chain, so will not trigger any tests
+  not compilation when a Pull Request is submitted to this branch
+- The README.md file from master branch is modified as follow:
+
+::
+
+  FRRouting XXX Development Branch
+  ================================
+
+  This special FRRouting branch is dedicated to the development of XXX.
+  The code located under this branch is a **Work In Progress**. The code may
+  compile or not, may run or not. In any case, this code **MUST NOT BE USED IN
+  PRODUCTION**.
+
+  FRRouting
+  ---------
+
+  FRR is free software that implements a ...
+
+
+Process to Request Feature Branch
+---------------------------------
+
+Only feature listed in FRR Wiki Features Requests
+(see https://github.com/FRRouting/frr/wiki/Feature-Requests) can be subject of
+this process.
+
+Everybody *(NOTE: another option is to limit this possibility to the mainteners
+only which will simplify the process, but close the door to not mainteners)*
+could ask for a new feature branch by following the bellow process:
+
+- **Prerequisit:** If feature is not listed in FRR Wiki Features Requests,
+  first create a new issue in FRR GitHub to request the new feature and wait
+  the feature is added to the Wiki
+- **Request:** [Send a simple mail is send to @frr-dev mailing list | Fill a
+  new issue | Update issue of the feature] *(NOTE: we need to choose the
+  appropriate method)* with a short description of the devlopment with the
+  issue number *(if mail method is choosen)*, and, if known, the list of
+  partners that express to work on. *(NOTE: these are the minimum information
+  I thought of. Of course additional information could be added)*
+- **Examination**: Request is disscussed during [the regular Tuesday meeting |
+  the TSC meeting] *(NOTE: we need to choose between the 2 options)*.
+- **Decision**: If request is approved, corresponding branch is created (see
+  below). A mail is sent to the @frr-dev mailing list to announce the creation
+  of the new branch and a link to that branch is added to the issue and the
+  FRR Wiki Features Requests page. If it is rejected, a mail with the reasons
+  of the rejection is also sent to the @frr-dev mailing list *(NOTE: if GitHub
+  issue method is choosen for the request, above text must be adapted, but,
+  I think that we could keep the announcement throuhg the mailing list)*.
+- **Creation**: If the request is approved, branch **dev-XXX** is created by
+  a TSC member or any personn that have the right to do it. A Owner of the
+  feature branch is designated as the responsible of this new feature branch.
+  The Owner is [a maintainers designated to shepherd the development on this
+  feature branch | the requester of the feature branch who is promoted as
+  maintainer to get commit rights on this branch] *(NOTE: we must decide if
+  we go for a shepherded maintainer or promote the requester if (s)he is not
+  already)*.
+
```

Comment:
I think a specific web page should be done, that contains the details of the dev branch ( partners, etc..). that will clarify what will be done and what is the status of that branch.
it will be up to the shepherd to keep up to date the wiki page.